File: src/content/design-patterns/factory/php/code.php
<?php

enum PaymentSystemType: string {
    case PAYPAL = 'paypal';
    case WEBMONEY = 'webmoney';
    case CAT_BANK = 'cat-bank';
}





// Выполните операцию payment для платежной системы `type`
function payment(PaymentSystemType $type, int $base, int $amount): int {

}

File: content/design-patterns/factory/src/go/code.go
package factory

type PaymentSystemType string

const (
	PayPal   PaymentSystemType = "paypal"
	WebMoney PaymentSystemType = "webmoney"
	CatBank  PaymentSystemType = "cat-bank"
)

func Payment(tp PaymentSystemType, base int, amount int) int {
	panic("Not implemented")
}
